src/functions-and-arrays.js

```diff
@@ -29,9 +149,47 @@ const wordsUnique = [
   'bring'
 ];
 
+
+const uniquifyArray = (array) => {
+  if (array.length === 0) {
+    return null;
+  }
+
+  let uniqueArray = [];
+
+  for (let i = 0; i < array.length; i++){
+    let word = array[i];
+    if (array.indexOf(word, i+1) === -1) {
+      uniqueArray.push(word);
+    } 
+  };
+
+  return uniqueArray;
+};
+
+console.log(uniquifyArray(wordsUnique));
+
 // Iteration #6: Find elements
 const wordsFind = ['machine', 'subset', 'trouble', 'starting', 'matter', 'eating', 'truth', 'disobedience'];
 
+const doesWordExist = (array, word) => {
+  if (array.length === 0) {
+    return null;
+  }
+
+  for (let i = 0; i < array.length; i++) {
+    let currentWord = array[i];
+    let lastIndex = array.length -1;
+    if (currentWord === word ) {
+      return true;
+      break;
```

Comment:
Try to avoid using "break", it is necessary in the switch loops but you shouldn't use it anywhere else (maybe in super advanced stuff sometimes, but in the bootcamp I am sure you won't need it anywhere but in the switch loops).

Actually, when you delete this break your code still works.